streams/src/main/java/org/apache/kafka/streams/processor/internals/SourceNode.java

```diff
@@ -29,14 +30,20 @@
     private ProcessorContext context;
     private Deserializer<K> keyDeserializer;
     private Deserializer<V> valDeserializer;
+    private final TimestampExtractor timestampExtractor;
 
-    public SourceNode(String name, List<String> topics, Deserializer<K> keyDeserializer, Deserializer<V> valDeserializer) {
+    public SourceNode(String name, List<String> topics, TimestampExtractor timestampExtractor, Deserializer<K> keyDeserializer, Deserializer<V> valDeserializer) {
```

Comment:
I recognized this issue:

       ` builder.stream(Pattern.compile("t.*")); `   // this adds sourceNode with name "t*", (before this PR it was `"Pattern[t*]"`)
       After that, in `StreamTask:120` when one executes:

         `   final SourceNode source = topology.source(partition.topic());`   // where `partition.topic()` is "t1"

It returns `null` because  `topology.source()` is just key value lookup. 
So if the source is added by pattern I moved extra extra lookup from `StreamTask` to` topology.source()`